Sync/Synchronizers/Bookmarks/BookmarksSynchronizer.swift

```diff
@@ -361,12 +365,16 @@ class ThreeWayBookmarksStorageMerger: BookmarksStorageMerger {
                     return self.applyLocalDirectlyToMirror()
                 case (false, true):
                     // No outgoing changes. Unilaterally apply remote changes if they're consistent.
-                    return self.buffer.validate() >>> self.applyIncomingDirectlyToMirror
+                    return self.buffer.validate().bind { result in
+                        result.isSuccess ? self.applyIncomingDirectlyToMirror() : deferMaybe(result.failureValue ?? BookmarksMergeConsistencyError())
```

Comment:
Again, if validation fails, it'll fill the deferred with a failure, and won't move on to the next step. Why do you need `bind`?